loki/batch/scheduler.py

```diff
@@ -545,6 +545,8 @@ def _get_definition_items(_item, sgraph_items):
             if transformation.renames_items or transformation.creates_items:
                 kwargs['item_factory'] = self.item_factory
                 kwargs['scheduler_config'] = self.config
+                kwargs['graph'] = graph
```

Comment:
We need to consider whether we want to directly provide the SGraph to transformations or build an API to semantically capture the changes and have the scheduler apply them afterwards.
The latter seems more elegant (not hacking with core data structures) but it might be tricky to build something equally flexible.
Also note that tthe `graph` may be a temporary object here when traversing a file graph. In that case, this would not work at all.